source/dub/compilers/utils.d

```diff
@@ -73,6 +73,50 @@ unittest {
 }
 
 
+/**
+	Determines if a specific file name has the extension related to dynamic libraries.
+
+	This includes dynamic libraries and for Windows pdb, export and import library files.
+*/
+bool isDynamicLibraryFile(const scope ref BuildPlatform platform, string f)
+{
+	import std.path;
+	switch (extension(f)) {
+		default:
+			return false;
+		case ".lib", ".pdb", ".dll", ".exp":
+			return platform.platform.canFind("windows");
```

Comment:
I think we should make a function in BuildPlatform like `isWindows` which checks for this, hardcoding the string everywhere seems kinda bad.